app/models/data_request.rb

```diff
@@ -30,13 +45,33 @@ def kase
 
   private
 
+  def validate_from_date_before_to_date
+    if date_from.present? && date_to.present? && date_from >= date_to
```

Comment:
I initially read this as if the date was the "present", as in now? Was thinking how can these dates be the the same haha 🤦

Reply:
Maybe this could be wrapped in a `dates_submitted?` method or something.

```ruby
def dates_submitted?
  dates_from.present? && date_to.present?
end

# then:

if dates_submitted? && date_from >= date_to
```

I think this is really just me misinterpreting things at first glance more than anything, but maybe something to think about.

Dates are odd in that `date.past?` (don't thing this is a real method) `date.present?` have sort of ambiguous names as they can relate to "existence" in the latter case, and a "place in time" for both.

Reply:
Good spot, thanks! It was annoying me about the same date thing too. I completely see your point about the English language of .present?, have done something along these lines, nice idea!